src/Microsoft.AspNetCore.Proxy/ProxyMiddleware.cs

```diff
@@ -59,8 +63,51 @@ public ProxyMiddleware(RequestDelegate next, IOptions<ProxyOptions> options)
             _httpClient = new HttpClient(_options.BackChannelMessageHandler ?? new HttpClientHandler());
         }
 
-        public async Task Invoke(HttpContext context)
+        public Task Invoke(HttpContext context) => context.WebSockets.IsWebSocketRequest ? HandleWebSocketRequest(context) : HandleHttpRequest(context);
+
+        private async Task HandleWebSocketRequest(HttpContext context)
+        {
+            using (var client = new ClientWebSocket())
+            {
+                foreach (var kv in context.Request.Headers)
+                {
+                    if (!NotForwardedWebSocketHeaders.Contains(kv.Key))
+                    {
+                        client.Options.SetRequestHeader(kv.Key, kv.Value);
+                    }
+                }
+
+                var wsScheme = string.Equals(_options.Scheme, "https", StringComparison.OrdinalIgnoreCase) ? "wss" : "ws";
+                var uriString = $"{wsScheme}://{_options.Host}:{_options.Port}{context.Request.PathBase}{context.Request.Path}{context.Request.QueryString}";
+
+                await client.ConnectAsync(new Uri(uriString), CancellationToken.None);
```

Comment:
I agree about error handling. Fixed it. However full implementation requires more than netstandard13 as WebException is not available there. So I created a generic implementation with 400 for start. After we figure out what to do with this WebException I can uncomment and add full error handling with entire HttpResponse forwarding.